subiquity/models/mirror.py

```diff
@@ -32,12 +36,111 @@
 
 log = logging.getLogger('subiquity.models.mirror')
 
+DEFAULT_SUPPORTED_ARCHES_URI = "http://archive.ubuntu.com/ubuntu"
+DEFAULT_PORTS_ARCHES_URI = "http://ports.ubuntu.com/ubuntu-ports"
+
+LEGACY_DEFAULT_PRIMARY_SECTION = [
+    {
+        "arches": PRIMARY_ARCHES,
+        "uri": DEFAULT_SUPPORTED_ARCHES_URI,
+    }, {
+        "arches": ["default"],
+        "uri": DEFAULT_PORTS_ARCHES_URI,
+    },
+]
 
 DEFAULT = {
     "preserve_sources_list": False,
 }
 
 
+@attr.s(auto_attribs=True)
+class PrimaryElement(abc.ABC):
+    parent: "MirrorModel" = attr.ib(kw_only=True)
+
+    def stage(self) -> None:
+        self.parent.primary_staged = self
+
+    def elect(self) -> None:
+        self.parent.primary_elected = self
+
+    @abc.abstractmethod
+    def serialize_for_ai(self) -> Any:
+        """ Serialize the element for autoinstall. """
+
+
+@attr.s(auto_attribs=True)
+class PrimaryEntry(PrimaryElement):
+    # Having uri set to None is only valid for a country mirror.
+    uri: Optional[str] = None
+    arches: Optional[List[str]] = None
```

Comment:
I see the unittest for this tests both arches == [] and None. What do they both mean?

Reply:
* `arches = []` means that no arches are supported. (I hope we don't see this in a real world scenario :+1: ).
* `arches = None` means that the current CPU architecture is supported.

I will make sure to add more documentation!

Reply:
Updated. Also fixed issue causing the list of arches passed to curtin to be empty. Since we only give curtin a single URI, passing arches=["default"] ensures it uses it and does not revert to http://archive.ubuntu.com